### PR TITLE
Vrrpe pynos

### PR DESCRIPTION
**Pull request by geethagadipuuri**
Vrrpe pynos calls for 
1. Vrrpe
2. Vrrpe_vip
   3.Vrrpe_vmac

### CODE REVIEW

File: pynos/versions/base/interface.py [hunk 1]
```diff
@@ -421,10 +420,10 @@ def ip_address(self, **kwargs):
         method_name = None
         method_class = self._interface
         if ipaddress.version == 4:
```
Comment (tnadeau) [inline pynos/versions/base/interface.py:422]: probably best to have a constant defined for any literals - same with the 6 below.

File: pynos/versions/base/interface.py [hunk 2]
```diff
@@ -421,10 +420,10 @@ def ip_address(self, **kwargs):
...
-            method_name = 'interface_%s_ip_ip_config_address_'\
+            method_name = 'interface_%s_ip_ip_config_address_' \
                           'address' % int_type
         elif ipaddress.version == 6:
```
Comment (tnadeau) [inline pynos/versions/base/interface.py:425]: same RE: literals

File: pynos/versions/base/interface.py [hunk 3]
```diff
@@ -1866,10 +1865,10 @@ def vrrp_vip(self, **kwargs):
 
         if ipaddress.version == 4:
             vrrp_args['version'] = '3'
```
Comment (tnadeau) [inline pynos/versions/base/interface.py:1867]: literal use - should there be an enum for this?

**Review by tnadeau — CHANGES_REQUESTED** [review]
a little tidying up of the code has been requested.

File: pynos/versions/base/interface.py [hunk 4]
```diff
@@ -2926,3 +2925,251 @@ def get_port_chann_detail_request(last_aggregator_id):
 
     def bfd(self, **kwargs):
         raise NotImplementedError
+
```
Comment (samirsss) [inline pynos/versions/base/interface.py:2928]: Why is this method added again? I thought we already have a method like enable_switchport or something similar

File: pynos/versions/base/interface.py [hunk 5]
```diff
@@ -2926,3 +2925,251 @@ def get_port_chann_detail_request(last_aggregator_id):
...
+                             'physical interfaces or x for port channel.')
+
+        mode_args = dict(name=name, vlan_mode=mode)
+        switchport_mode = getattr(self._interface, 'interface_%s_switchport_'
+                                                   'mode_vlan_mode' % int_type)
+        config = switchport_mode(**mode_args)
+        print pynos.utilities.print_xml_string(config)
+        return callback(config)
+
+    def vrrpe_vip(self, **kwargs):
```
Comment (samirsss) [inline pynos/versions/base/interface.py:2989]: can we add support for Get and Delete, similar to other pynos methods?

File: pynos/versions/base/interface.py [hunk 6]
```diff
@@ -2926,3 +2925,251 @@ def get_port_chann_detail_request(last_aggregator_id):
...
+
+        Examples:
+            >>> import pynos.device
+            >>> switches = ['10.24.39.211', '10.24.39.203']
+            >>> auth = ('admin', 'password')
+            >>> for switch in switches:
+            ...     conn = (switch, '22')
+            ...     with pynos.device.Device(conn=conn, auth=auth) as dev:
+            ...         output = dev.interface.anycast_mac(rbridge_id='225',
+            ...         mac='aabb.ccdd.eeff', delete=True)
```
Comment (samirsss) [inline pynos/versions/base/interface.py:3019]: Is creation of anycast_mac needed for this doctest?

File: pynos/versions/base/interface.py [hunk 7]
```diff
@@ -2926,3 +2925,251 @@ def get_port_chann_detail_request(last_aggregator_id):
...
+            ...         ip_addr='2002:4818:f000:1ab:cafe:beef:1000:2/64')
+            ...         dev.interface.vrrpe_vip(int_type='ve', name='89',
+            ...         vrid='1', vip='172.16.1.2/24', rbridge_id='225')
+            ...         dev.interface.vrrpe_vip(int_type='ve', name='89',
+            ...         vrid='1', vip='fe80::dafe:beef:1000:1/64',
+            ...         rbridge_id='225')
+            ...         dev.interface.vrrpe_vip(int_type='ve', name='89',
+            ...         vrid='1', vip='2002:4818:f000:1ab:cafe:beef:1000:1/64',
+            ...         rbridge_id='225')
+            ...         output = dev.services.vrrpe(ip_version='6',
```
Comment (samirsss) [inline pynos/versions/base/interface.py:3052]: You pass in delete=True/False but I dont see support for it in the code? Has this doctest been tested?

File: pynos/versions/base/interface.py [hunk 8]
```diff
@@ -2926,3 +2925,251 @@ def get_port_chann_detail_request(last_aggregator_id):
...
+            ...         name='89',name='89',vrid='1',
+            ...         vip='2002:4818:f000:1ab:cafe:beef:1000:1/64',
+            ...         rbridge_id='225')
+            ...         output = dev.services.vrrpe(ip_version='6',
+            ...         enabled=False, rbridge_id='225')
+            ...         output = dev.services.vrrpe(enabled=False,
+            ...         rbridge_id='225')
+            ...         output = dev.interface.vrrpe_vmac(int_type='ve',
+            ...         name='89', vrid='1', rbridge_id='1',
+            ...         virtual_mac='aaaa.bbbb.cccc')
```
Comment (samirsss) [inline pynos/versions/base/interface.py:3140]: Lets add support for get and delete operation here for the virtual mac as well

File: pynos/versions/base/services.py [hunk 9]
```diff
@@ -93,3 +94,55 @@ def vrrp(self, **kwargs):
...
+        enabled = kwargs.pop('enabled', True)
+        rbridge_id = kwargs.pop('rbridge_id', '1')
+        callback = kwargs.pop('callback', self._callback)
+        vrrpe_args = dict(rbridge_id=rbridge_id)
+        vrrpe_method = 'rbridge_id_protocol_hide_vrrp_holder_vrrp_extended'
+        if ip_version == '6':
+            vrrpe_method = 'rbridge_id_ipv6_proto_vrrpv3_vrrp_extended'
+        vrrpe = getattr(self._rbridge, vrrpe_method)
+        config = vrrpe(**vrrpe_args)
+        if not enabled:
```
Comment (samirsss) [inline pynos/versions/base/services.py:146]: What's the mechanism to check if vrrpe is enabled or disabled?

**Review by samirsss — CHANGES_REQUESTED** [review]

**Comment by geethagadipuuri** [comment]
closing this as new request was created addressing the changes

**Pull request closed without merging**